pkg/apis/v1alpha1/cluster_template.go

```diff
@@ -68,6 +68,20 @@ type TemplateSpec struct {
 	// stamped by this template
 	// +optional
 	HealthRule *HealthRule `json:"healthRule,omitempty"`
+
+	// Lifecycle specifies whether template modifications should result in originally
+	// created objects being updated (`default`) or in new objects created alongside
+	// original objects (`immutable` or `tekton`).
+	// +kubebuilder:validation:Enum=default;immutable;tekton
+	// +kubebuilder:default="default"
```

Comment:
"Default" lifecycle is only meaningful if you know what things used to look like. I think "mutable" is a better, specific term.

Reply:
I'd say this is also a design question for @zrob

Reply:
agree default doesn't help much. let's go with "mutable" as the default behavior.

also propose we remove immutable. it is confusing to have three options and only two behaviors. we're being explicit with tekton right now. we can add immutable and give it meaning if we have a specific use case to deliver.

lifecycle: mutable (default), tekton

Reply:
lifecycle `default` has been changed to `mutable`.